### PR TITLE
Feat(builder unlock): Upgrade contract

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

File: contracts/builder_unlock/Cargo.toml [hunk 1]
```diff
@@ -20,10 +20,10 @@ cw-storage-plus = {version = "0.8.0", features = ['iterator']}
 schemars = "0.8.1"
 serde = { version = "1.0.103", default-features = false, features = ["derive"] }
 astroport-governance = { path = "../../packages/astroport-governance", default-features = false, version = "1.0.0"}
+astroport = {git = "https://github.com/astroport-fi/astroport.git", branch = "master"}
```
Comment (epanchee) [inline contracts/builder_unlock/Cargo.toml:23]: Correct git path is "https://github.com/astroport-fi/astroport-core". We use main branch as default.

File: contracts/builder_unlock/src/contract.rs [hunk 2]
```diff
@@ -551,8 +761,48 @@ fn query_simulate_withdraw(
...
+                    },
+                )?;
+
+                let keys = STATUSV100
+                    .keys(deps.storage, None, None, cosmwasm_std::Order::Ascending {})
+                    .map(|v| String::from_utf8(v).map_err(StdError::from))
+                    .collect::<Result<Vec<String>, StdError>>()?;
+
+                for key in keys {
+                    let status_v110 = STATUSV100.load(deps.storage, &Addr::unchecked(&key))?;
```
Comment (epanchee) [inline contracts/builder_unlock/src/contract.rs:786]: Rename to ```status_v100``` for consistency.

File: packages/astroport-governance/src/builder_unlock.rs [hunk 3]
```diff
@@ -18,13 +18,16 @@ pub struct State {
...
+    /// Amount of ASTRO tokens deposited into contract but without allocation
+    pub unallocated_tokens: Uint128,
 }
 
 impl Default for State {
     fn default() -> Self {
         State {
             total_astro_deposited: Uint128::zero(),
             remaining_astro_tokens: Uint128::zero(),
+            unallocated_tokens: Uint128::zero(),
```
Comment (epanchee) [inline packages/astroport-governance/src/builder_unlock.rs:30]: These are defaults for Uint128. I suggest to remove this impl block and use ```#[derive(..., Default)]``` for State struct.

File: packages/astroport-governance/src/builder_unlock.rs [hunk 4]
```diff
@@ -70,12 +73,15 @@ impl Default for AllocationParams {
...
     pub astro_withdrawn: Uint128,
+    /// Already unlocked amount after decreasing
+    pub unlocked_amount_checkpoint: Uint128,
 }
 
 impl Default for AllocationStatus {
     fn default() -> Self {
         AllocationStatus {
             astro_withdrawn: Uint128::zero(),
+            unlocked_amount_checkpoint: Uint128::zero(),
```
Comment (epanchee) [inline packages/astroport-governance/src/builder_unlock.rs:84]: Same as I commented above

File: packages/astroport-governance/src/builder_unlock.rs [hunk 5]
```diff
@@ -185,5 +207,7 @@ pub mod msg {
         pub total_astro_deposited: Uint128,
         /// Currently available ASTRO tokens that weren't yet withdrawn from the contract
         pub remaining_astro_tokens: Uint128,
+        /// Currently available ASTRO tokens to withdraw or increase allocations be the owner
```
Comment (epanchee) [inline packages/astroport-governance/src/builder_unlock.rs:210]: by the owner

File: contracts/builder_unlock/Cargo.toml (hunk 1 above)
Comment (maksk7) [reply contracts/builder_unlock/Cargo.toml]: Changed

File: contracts/builder_unlock/src/contract.rs (hunk 2 above)
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs]: Ok, done, thanks

File: packages/astroport-governance/src/builder_unlock.rs (hunk 3 above)
Comment (maksk7) [reply packages/astroport-governance/src/builder_unlock.rs]: Done

File: packages/astroport-governance/src/builder_unlock.rs (hunk 4 above)
Comment (maksk7) [reply packages/astroport-governance/src/builder_unlock.rs]: Done, thanks

File: packages/astroport-governance/src/builder_unlock.rs (hunk 5 above)
Comment (maksk7) [reply packages/astroport-governance/src/builder_unlock.rs]: Changed

**Review by epanchee — APPROVED** [review]

**Review by P-Yevhenii — APPROVED** [review]

File: contracts/builder_unlock/src/contract.rs [hunk 6]
```diff
@@ -380,6 +409,184 @@ fn execute_drop_new_receiver(deps: DepsMut, _env: Env, info: MessageInfo) -> Std
...
+
+            PARAMS.save(deps.storage, &receiver, &params)?;
+            STATE.save(deps.storage, &state)?;
+        }
+        None => {
+            return Err(StdError::generic_err("Proposed receiver not set"));
+        }
+    }
+
+    Ok(Response::new())
```
Comment (ShadoySV) [inline contracts/builder_unlock/src/contract.rs:528]: Let's add some useful attributes

File: contracts/builder_unlock/src/contract.rs [hunk 7]
```diff
@@ -380,6 +409,184 @@ fn execute_drop_new_receiver(deps: DepsMut, _env: Env, info: MessageInfo) -> Std
...
+/// * **_info** is an object of type [`MessageInfo`].
+///
+/// * **receiver** is an object of type [`String`]. Increasing receiver.
+///
+/// * **amount** is an object of type [`Uint128`]. ASTRO amount to increase.
+///
+/// * **deposit_amount** is an [`Option`] of type [`Uint128`]. Amount of ASTRO to increase using CW20 Receive.
+fn execute_increase_allocation(
+    deps: DepsMut,
+    _env: Env,
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs]: Removed, thanks

File: contracts/builder_unlock/src/contract.rs (hunk 6 above)
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs]: Ok, thanks

File: contracts/builder_unlock/src/contract.rs [hunk 8]
```diff
@@ -380,6 +409,184 @@ fn execute_drop_new_receiver(deps: DepsMut, _env: Env, info: MessageInfo) -> Std
...
+    let unlocked_amount = compute_unlocked_amount(
+        env.block.time.seconds(),
+        params.amount,
+        &params.unlock_schedule,
+        status.unlocked_amount_checkpoint,
+    );
+    let astro_withdrawable = unlocked_amount - status.astro_withdrawn;
+    let locked_amount = params.amount - astro_withdrawable;
+
+    if locked_amount < amount {
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs:460]: Fixed, thanks

**Pull request merged**